tests/EvmAcceptanceTests/test/Subscriptions.js

```diff
@@ -18,13 +12,23 @@ describe("Subscriptions functionality", function () {
     const Contract = await ethers.getContractFactory("Subscriptions");
     senderAddress = (await ethers.getSigner()).address;
     contract = await Contract.deploy();
+  });
+
+  beforeEach(async function () {
     provider = new ethers.providers.WebSocketProvider(general_helper.getWebsocketUrl());
     eventsContract = new ethers.Contract(contract.address, contract.interface, provider);
   });
+
   afterEach(async function () {
     eventsContract.removeAllListeners();
   });
 
+  // Ensures all events in current eventloop run are dispatched
+  async function waitForEvents(events, timeout = 5000) {
```

Comment:
does it work? shouldn't you change all calls to this function to `this.waitForEvents`?

Reply:
Nope, it looks that the scope is not private and it works without `this`.